distance-metrics/src/lib.rs

```diff
@@ -0,0 +1,13 @@
+pub mod legacy;
+pub mod metric;
+pub mod simple;
+pub mod types;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+pub mod simple_sse;
+
+#[cfg(target_arch = "x86_64")]
+pub mod simple_avx;
+
+#[cfg(target_arch = "aarch64")]
+pub mod simple_neon;
```

Comment:
Style suggestions: my guidelines for Rust module files is that the ideal size is a few hundred lines. To me, spreading this stuff over multiple modules makes the code harder to follow rather than easier. I think it makes sense to keep the different SIMD implementation in their own modules because they need the most code, but I'd prefer to move everything from `legacy`, `metric` and `types` just into `lib.rs`.